src/main/java/uk/ac/cam/cl/dtg/util/email/MailJetApiClientWrapper.java

```diff
@@ -0,0 +1,181 @@
+                .build();
+
+        this.mailjetClient = new MailjetClient(options);
+        this.newsListId = properties.getProperty(MAILJET_NEWS_LIST_ID);
+        this.eventsListId = properties.getProperty(MAILJET_EVENTS_LIST_ID);
+    }
+
+    /**
+     *  Get user details for an existing MailJet account
+     *
+     * @param mailjetIdOrEmail - email address or MailJet user ID
+     * @return JSONObject of the MailJet user
+     * @throws MailjetException  - if underlying MailjetClient throws an exception
+     */
+    public JSONObject getAccountByIdOrEmail(final String mailjetIdOrEmail) throws MailjetException {
+        if (null == mailjetIdOrEmail) {
+            return null;
+        }
+        MailjetRequest request = new MailjetRequest(Contact.resource, mailjetIdOrEmail);
+        MailjetResponse response = mailjetClient.get(request);
+        JSONArray responseData = response.getData();
+        if (response.getTotal() == 1) {
+            return responseData.getJSONObject(0);
+        }
+        return null;
+    }
+
+    /**
+     *  Perform an asynchronous GDPR-compliant deletion of a MailJet user.
+     *
+     * @param mailjetId - MailJet user ID
+     * @throws MailjetException  - if underlying MailjetClient throws an exception
+     */
+    public void permanentlyDeleteAccountById(final String mailjetId) throws MailjetException {
+        Validate.notNull(mailjetId);
+        MailjetRequest request = new MailjetRequest(Contacts.resource, mailjetId);
+        MailjetResponse response = mailjetClient.delete(request);
+        int responseStatus = response.getStatus();
+        if (!(responseStatus == Response.Status.OK.getStatusCode())) {
+            // TODO: Do we want to get any of the data from this request?
+            throw new MailjetException("Failed to delete user!");
+        }
+    }
+
+    /**
+     *  Add a new user to MailJet or find by email if they already exist as a fallback
+     *
+     * @param email - email address
+     * @return the MailJet user ID
+     * @throws MailjetException  - if underlying MailjetClient throws an exception
+     */
+    public String addNewUser(final String email) throws MailjetException {
+        if (null == email) {
+            return null;
+        }
+        try {
+            MailjetRequest request = new MailjetRequest(Contact.resource).property(Contact.EMAIL, email);
+            MailjetResponse response = mailjetClient.post(request);
+            // Get MailJet ID out:
+            JSONObject responseData = response.getData().getJSONObject(0);
+            return Integer.toString(responseData.getInt("ID"));
+        } catch (MailjetClientRequestException e) {
+            if (e.getMessage().contains("already exists")) {
+                // FIXME - we need to test that this response always comes back with "already exists" in the message
+                log.warn(String.format("Attempted to create a user with email (%s) that already existed!", email));
```

Comment:
In this event should we throw an exception or is the consumer likely not to care? In which case do we want to modify the method name to indicate it is addOrGetUser?

Reply:
I was just trying to make it idempotent in case things went wrong, or the user has already been synced. Happy to rename it!